plugins/compliance-hub/frontend/public/javascripts/countly.views.js

```diff
@@ -13,6 +13,55 @@ window.ConsentManagementView = countlyView.extend({
                 countlyConsentManager.initialize()).then(function () {});
 		}
     },
+    getExportAPI: function(tableID){
+        if(tableID === 'd-table-users'){
+            var requestPath = '/o/app_users/consents?api_key='+countlyGlobal.member.api_key + 
+            "&app_id=" + countlyCommon.ACTIVE_APP_ID +  "&iDisplayStart=0"
+            var apiQueryData = {
+                api_key: countlyGlobal.member.api_key,
+                app_id: countlyCommon.ACTIVE_APP_ID,
+                path: requestPath,
+                method: "GET",
+                filename:"Compliance_users_on_" + moment().format("DD-MMM-YYYY"),
+                prop: ['aaData']
+            };
+            return apiQueryData;
+        }
+        if(tableID === 'd-table-actionlogs') {
+            var query = app.activeView.action_query || {a:{$in:["export_app_user","app_user_deleted","export_app_user_deleted"]}};
+            query["i.app_id"] = countlyCommon.ACTIVE_APP_ID;
+
+            var requestPath = '/o?api_key='+countlyGlobal.member.api_key + 
+            "&app_id=" + countlyCommon.ACTIVE_APP_ID +  "&method=systemlogs&iDisplayStart=0" +
+            "&query="+ encodeURIComponent(JSON.stringify(query)) + 
+            "&period=" + countlyCommon.getPeriodForAjax()
+            var apiQueryData = {
+                api_key: countlyGlobal.member.api_key,
+                app_id: countlyCommon.ACTIVE_APP_ID,
+                path: requestPath,
+                method: "GET",
+                filename:"Compliance_export_or_purge_history_on_" + moment().format("DD-MMM-YYYY"),
+                prop: ['aaData']
+            };
+            return apiQueryData;
+        }
```

Comment:
This table was actually added by systemlogs, would make sense that export was handled by systemlogs as well

Reply:
But the table defined the same file here, so I think we should keep the download query related function with table definition. How do you think?

Reply:
No, table is defined in systemlogs plugin:
https://github.com/Countly/countly-server/blob/master/plugins/systemlogs/frontend/public/javascripts/countly.views.js#L349